aws/resource_aws_wafregional_sql_injection_match_set.go

```diff
@@ -0,0 +1,233 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/waf"
+	"github.com/aws/aws-sdk-go/service/wafregional"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsWafRegionalSqlInjectionMatchSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafRegionalSqlInjectionMatchSetCreate,
+		Read:   resourceAwsWafRegionalSqlInjectionMatchSetRead,
+		Update: resourceAwsWafRegionalSqlInjectionMatchSetUpdate,
+		Delete: resourceAwsWafRegionalSqlInjectionMatchSetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
Something up for discussion: should we automatically be adding `ValidateFunc: validation.NoZeroValues,` for `Required: true` and `Type: schema.TypeString` where it makes sense?

Reply:
Hm, I'm not sure - it sounds like something that should be solved at higher (schema) level. I mean if from provider-dev perspective I say "it's required" I'd just assume that's enough to say there are no empty strings allowed.

Maybe there are edge cases when we do _not_ want that though? 🤷‍♂️

Reply:
I agree that it likely _should_ be fixed upstream 😄  The only weird edge cases I can think of are module usage somehow, but even that sounds pretty far fetched.